perf/benchmark/runner/generate_policies/generate_polices.go

```diff
@@ -17,47 +17,81 @@ package main
 import (
     "fmt"
     "bytes"
+    "github.com/ghodss/yaml"
+    "encoding/json"
+
+    authzpb "istio.io/api/security/v1beta1"
 )
 
 type ruleOption struct {
     occurance int
     g         generator
 }
 
-func generateAuthorizationPolicy(action string, ruleToOccurences map[string]*ruleOption) (string, error) {
-    yaml := bytes.Buffer{}
-    yaml.WriteString("spec:\n")
-    yaml.WriteString(" action: " + action + "\n")
+type MyPolicy struct {
+    ApiVersion string `json:"apiVersion"`
+    Kind       string `json:"kind"`
+    Metadata   MetadataStruct `json:"metadata"`
+    Spec       authzpb.AuthorizationPolicy `json:"spec"`
+}
 
-    if (len(ruleToOccurences) == 0) {
-        yaml.WriteString(" {}")
-    } else {
-        yaml.WriteString(" rules:\n")
+type MetadataStruct struct {
+    Name      string  `json:"name"`
+    Namespace string  `json:"namespace"`
+}
+
+func formYaml(policy *MyPolicy) (string, error) {
+    header, err := json.Marshal(policy)
+    if err != nil {
+        return "", err
+    }
+
+    yaml, err := yaml.JSONToYAML([]byte(header))
+    if err != nil {
+        return "", err
     }
 
-    for rule, ruleOp := range ruleToOccurences {
-        yaml.WriteString("  " + rule + ":\n")
-        rules, err := ruleOp.g.generate(rule, ruleOp.occurance)
+    return string(yaml), nil
+}
+
+func generateAuthorizationPolicy(action string, ruleToOccurences map[string]*ruleOption, policy *MyPolicy) (string, error) {
+    spec := &authzpb.AuthorizationPolicy{}
+    // This action will be set by the paramater action
+    spec.Action = 1;
```

Comment:
I am not sure why but when assigned the Action the only value that is read is 1 which results in a DENY action. 0 should create a ALLOW action but that creates no action

Reply:
there should be a constant for deny and allow, check the authzpb package for that.